services/connector/api_test.go

```diff
@@ -0,0 +1,80 @@
+package connector
+
+import (
+	"database/sql"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+
+	gethrpc "github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/status-im/status-go/appdatabase"
+	"github.com/status-im/status-go/params"
+	statusRPC "github.com/status-im/status-go/rpc"
+	"github.com/status-im/status-go/t/helpers"
+	"github.com/status-im/status-go/transactions/fake"
+)
+
+func createDB(t *testing.T) (*sql.DB, func()) {
+	db, cleanup, err := helpers.SetupTestSQLDB(appdatabase.DbInitializer{}, "provider-tests-")
+	require.NoError(t, err)
+	return db, func() { require.NoError(t, cleanup()) }
+}
+
+func setupTestAPI(t *testing.T) (*API, func()) {
+	t.Skip("skip test using infura")
+	db, cancel := createDB(t)
+
+	txServiceMockCtrl := gomock.NewController(t)
+	server, _ := fake.NewTestServer(txServiceMockCtrl)
+	upstreamConfig := params.UpstreamRPCConfig{
+		URL:     "https://mainnet.infura.io/v3/800c641949d64d768a5070a1b0511938",
```

Comment:
is this token ok to share?

Reply:
Yes... That command creates a dummy upstream config. The same is reused in the api_test.go files for ens service for instance, or for web3provider for consistency.